### PR TITLE
meta: Add OWNERS file and PR/Issue templates

### PR DESCRIPTION
**Pull request by jbw976**
This PR adds some metadata files to this repo including the OWNERS.md and templates for PRs and issues.

I have also updated the access to this repo to match the proposed OWNERS, and added branch protection rules.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+can uncomment the below line to indicate which issue your PR fixes, for example
+"Fixes #500":
+
+-->
+Fixes #
+
+I have:
+
+- [ ] Read and followed Crossplane's [contribution process].
+- [ ] Run `make reviewable test` to ensure this PR is ready for review.
```
Comment (jbw976) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: there isn't yet a `reviewable` target, is there?

Comment (hasheddan) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: Not yet :)

**Review by hasheddan — APPROVED** [review]
LGTM! Thanks @jbw976! Were you wanting to fix that PR template? If not, I think we will add reviewable soon so I would be fine with going ahead and merging 👍

**Comment by jbw976** [comment]
I think it's OK to keep `reviewable` in the PR template as forward looking and optimistic, thanks @hasheddan! :)

**Pull request merged**